drivers/sensor/icm42670S/invn.algo.sw-integration.aml-42670S-gcc-zephyr-1.6.0/include/invn_algo_aml.h

```diff
@@ -0,0 +1,169 @@
+ *
+ *  @section AML_API API
+ *    The AIR MOTION Library uses the following functions :
+ *      @li invn_algo_aml_init() : This function must be called for library initialization, each time processing (re)starts.
+ *      @li invn_algo_aml_process() : This function must be called each time new sensors values are available. The library is designed to run at 100Hz.
+ *      @li invn_algo_aml_reset_swipe_recognition() : This function must be called to trigger each new swipe gesture detection.
+ *
+ *  @section AML_CALIBRATION Calibration
+ *    Gyroscope offsets values can fluctuate over time depending on various parameters.
+ *    The library supports a continuous calibration which computes gyroscope offsets values in real time.
+ *
+ *    Each time the invn_algo_aml_process() is called, gyroscope instant "staticness" is checked.
+ *    If the device is considered static, new gyroscope offset values are computed. The remote can then be considered as calibrated.
+ *    You can check 'InvnAlgoAMLOutput.status' to track calibration status over time.
+ *
+ *    'InvnAlgoAMLConfig.gyr_fsr' parameter is used to calibrate motion algorithms.
+ *    'InvnAlgoAMLConfig.acc_fsr' parameter is used for roll-compensation feature.
+ *
+ *  @section AML_ROLLCOMPENSATION RollCompensation
+ *    The Air Motion Library embeds a roll-compensation algorithm along with its pointing feature.
+ *    Roll-compensation allows user's movement to be reliably reproduced on the screen, independently from the device roll orientation.
+ *
+ *  @section AML_EASYCLICK EasyClick
+ *    This feature allows user to perform more accurate and stable mouse clicks for a better experience.
+ *    On click press, the library will freeze the pointer (dX and dY null) to avoid undesired movement.
+ *    During this period, if the device movement quantity is too important, the pointer will be released sooner.
+ *
+ */
+ 
+/* Define default algorithm init values */
+#define INVN_ALGO_AML_DELTA_GAIN_DEFAULT        15
+
+/* Define bit mask for output status */
+#define INVN_ALGO_AML_STATUS_DELTA_COMPUTED           0x1
+#define INVN_ALGO_AML_STATUS_STATIC                   0x2
+#define INVN_ALGO_AML_STATUS_NEW_GYRO_OFFSET          0x4
+#define INVN_ALGO_AML_STATUS_REMOTE_POSITION_CHANGE   0x8
+
+/* Define for different swipe detected */
+#define INVN_ALGO_AML_SWIPE_LEFT                0x1
+#define INVN_ALGO_AML_SWIPE_RIGTH               0x2
+#define INVN_ALGO_AML_SWIPE_UP                  0x4
+#define INVN_ALGO_AML_SWIPE_DOWN                0x8
+#define INVN_ALGO_AML_SWIPE_CLOCKWISE           0x10
+#define INVN_ALGO_AML_SWIPE_COUNTERCLOCKWISE    0x20
+
+
+/* Define for different remote position */
+typedef enum
+{
+	INVN_ALGO_AML_TOP,     //Z Down
+	INVN_ALGO_AML_BOTTOM,  //Z Up
+	INVN_ALGO_AML_LEFT,    //Y Up
+	INVN_ALGO_AML_RIGHT,   //Y Down
+	INVN_ALGO_AML_FRONT,   //X Up
+	INVN_ALGO_AML_REAR     //X Down
+}  RemotePosition;
+
+/* Forward declarations */
+struct inv_imu_device;
```

Comment:
Something is missing here??
I would expect something like: struct inv_imu_device **imu**;